Automattic/simplenote-ios
# Search: Dismissing Keyboard upon scroll

### Details:
In this PR we're making the SearchBar to resign its First Responder state, whenever you scroll over the Search Results.

Ref. #507 

@aerych Sorry about spamming you!!!

### Test: Dismiss upon Scroll
0. Set the `FeatureManager.advancedSearchEnabled` property to `true`
1. Log into your account
2. Tap over the Search Bar
3. Scroll over the Results

- [x] Verify the Keyboard is dismissed as soon as you begin dragging
- [x] Verify that **even** if the keyboard is hidden, the `Cancel` button remains enabled
- [x] Verify that pressing over the SearchBar triggers the keyboard again
- [x] Verify that pressing the `Cancel` button, while the Keyboard is hidden, effectively lets you exit Search Mode
- [x] Verify that you can get the SearchBar to become the first responder while the TableView remains inertia-scrolling (perform a quick swipe down / up!)


### Release
These changes do not require release notes.
